src/Payment.res

```diff
@@ -337,9 +395,15 @@ let make = (~paymentMode, ~integrateError, ~logger) => {
   }, (cardNumber, cvcNumber, cardExpiry, isCVCValid, isExpiryValid, isCardValid))
 
   React.useEffect(() => {
-    setCardError(_ => isCardValid->Option.getOr(true) ? "" : localeString.inValidCardErrorText)
+    setCardError(_ =>
+      isCardSupported->Option.getOr(true) && isCardValid->Option.getOr(true)
+        ? ""
+        : isCardSupported->Option.getOr(true)
+        ? localeString.inValidCardErrorText
+        : localeString.cardBrandConfiguredErrorText(cardBrand)
```

Comment:
Can we calcuate this value outside and use if else for better understanding of cases